### PR TITLE
Implementation copy operator for mkl-dnn

### PR DESCRIPTION
**Pull request by PenghuiCheng**
It is a operator to copy blob from ideep device to ideep device.

### CODE REVIEW

File: caffe2/ideep/operators/utility_ops.cc [hunk 1]
```diff
@@ -25,6 +25,23 @@ class CopyCPUToIDEEPOp final : public IDEEPOperator {
...
+  USE_SIMPLE_IDEEP_CTOR_DTOR(IDEEPCopyOp);
+  USE_IDEEP_DEF_ALIASES();
+
+  bool RunOnDevice() override {
+    const auto& X = OperatorBase::Input<itensor>(0);
+    auto* Y = Output(0);
+    if (X != *Y) {
+      Y->reinit_like(X);
+    }
+    ideep::direct_copy::compute(X, *Y);
```
Comment (yinghai) [inline caffe2/ideep/operators/utility_ops.cc]: Move this line into the if condition?

**Review by yinghai — COMMENTED** [review]
High level question: why do we need this?

File: caffe2/ideep/operators/utility_ops.cc (hunk 1 above)
Comment (PenghuiCheng) [reply caffe2/ideep/operators/utility_ops.cc]: it is a op which used in data_parallel_model.py. It support multi-devices mode.

Comment (PenghuiCheng) [reply caffe2/ideep/operators/utility_ops.cc]: yes,it should be moved to if condition.

Comment (PenghuiCheng) [reply caffe2/ideep/operators/utility_ops.cc]: This op is need by multi-devices mode for mkl-dnn,if we fallback this op,the performance will be affected。

**Comment by PenghuiCheng** [comment]
@yinghai  This op is need by multi-devices mode for mkl-dnn,if we fallback this op,the performance will be affected。And there are three test failing,but from log,it seems that has nothing to do with this patch. Could you review it?

File: caffe2/python/ideep/copy_op_test.py [hunk 2]
```diff
@@ -63,3 +64,23 @@ def test_copy_from_ideep_fallthrough(self):
...
+        c = randint(1, 64)
+        h = randint(1, 128)
+        w = randint(1, 128)
+        X = np.random.rand(n, c, h, w).astype(np.float32)
+        workspace.FeedBlob("X", X, self._get_deep_device())
+        workspace.RunOperatorOnce(op)
+        Y = workspace.FetchBlob("Y")
+        np.testing.assert_allclose(X, Y)
+
+if __name__ == "__main__":
```
Comment (yinghai) [inline caffe2/python/ideep/copy_op_test.py]: We don't need this.

**Review by yinghai — APPROVED** [review]

File: caffe2/python/ideep/copy_op_test.py (hunk 2 above)
Comment (PenghuiCheng) [reply caffe2/python/ideep/copy_op_test.py]: So,I need to remove the change of this file,is it right?

Comment (yinghai) [reply caffe2/python/ideep/copy_op_test.py]: Yes. Also please rebase with master and see if test failures will go away.

**Pull request closed without merging**